odoo/http.py

```diff
@@ -1938,7 +1938,7 @@ def set_csp(self, response):
             return
 
         mime, _params = cgi.parse_header(headers.get('Content-Type', ''))
-        if not mime.startswith('image/'):
+        if mime == 'image/svg+xml' or not mime.startswith('image/'):
```

Comment:
This would disable CSP for svg images, why would you need to disable CSP?

Reply:
Unlike other images, svgs are xml documents and their elements may contain styles.

Without this change, the browser renders a malformed image due to inline styles not being applied. (for instance, a barcode renders as a black square).

I'm not too familiar with CSP, though.
Maybe there's a way to bee a little bit more permisive with svgs, without fully disabling it?

Reply:
Indeed, that's what CSP is for, to forbid content injection from untrusted sources. Here we are enabling svg images only for barcodes which I suppose would be self-contained and plain-dark. Can you provide an example where disallowing inline-css on svg images makes it hard to read a barcode?

Reply:
Looking at the blog post you sent, maybe we should indeed craft a specific CSP for SVG

Reply:
Take for example the QR code generated from this URL (browser: Google Chrome):

`http://localhost:8069/report/barcode/QR/asd?format=svg&height=100&width=100`

**With the generic CSP rule, we get this:**
![image](https://user-images.githubusercontent.com/1914185/210274387-c50b9152-dc65-4c04-8cb6-d4a5492fa7dd.png)

**With the new svg-specific CSP rule (or none at all):**
![image](https://user-images.githubusercontent.com/1914185/210274344-f47cea98-a634-4e1e-8e44-b31cd7e16f91.png)

**By inspecting the svg element we can see the issue is some inline styles not being applied:**
<details>
<summary>See screenshot</summary>

![image](https://user-images.githubusercontent.com/1914185/210274459-bb6057ea-98ad-4166-92df-da6acc43d958.png)
</details>

---

Finally, and following some tips found online (that blog post included), I added the `style-src 'self' 'unsafe-inline'` rule only for `image/svg+xml`, and it's working nicely (at least for barcodes generated by reportlib)

Reply:
Is the unsafe-inline still required? [The bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1262842) seems solved with only `style-src 'self'`

Reply:
Actually, AFAICS that bug report is about Firefox incorrectly applying the CSP policy when rendering `svg` on an `img` tag (through the `src` attribute)

You see, rendering images through the `img` tag (`<img src="..." />`) is considered safe by the browsers, and thus the page's CSP is completely ignored (apart from the `image-src` rule, that must be valid) as it's deemed not necessary.

Apparently, Firefox had a bug where the SVG's inline styles were silently ignored when rendered using an `img` tag. That is solved, but it's not really our issue here.

We do need `'unsafe-inline'`, because we want styles that are inlined in the svg to be applied. There isn't any category/rule for "safe" styles, that rules blocks all inline styles. Although, in reality, they aren't really that unsafe (see for example https://github.com/angular/angular/issues/6361).

---

See my comment below for more